.tekton/.currency/scripts/generate_report.py

```diff
@@ -0,0 +1,243 @@
+# Standard Libraries
+import re
+from json import load
+
+# Third Party
+from requests import get
+from pandas import DataFrame
+from bs4 import BeautifulSoup
+from kubernetes import client, config
+
+JSON_FILE = "resources/table.json"
+REPORT_FILE = "docs/report.md"
+API_V1_ENDPOINT = "https://rubygems.org/api/v1/versions/"
+
+def filter_taskruns(taskrun_filter, taskruns):
+    filtered_taskruns = list(filter(taskrun_filter, taskruns))
+    filtered_taskruns.sort(
+        key=lambda tr: tr["metadata"]["creationTimestamp"], reverse=True
+    )
+
+    return filtered_taskruns
+
+
+def get_taskruns(namespace, task_name):
+    group = "tekton.dev"
+    version = "v1"
+    plural = "taskruns"
+
+    # access the custom resource from tekton
+    tektonV1 = client.CustomObjectsApi()
+    taskruns = tektonV1.list_namespaced_custom_object(
+        group,
+        version,
+        namespace,
+        plural,
+        label_selector=f"{group}/task={task_name}, triggers.tekton.dev/trigger=ruby-tracer-scheduled-pipeline-triggger",
+    )["items"]
+
+    return taskruns
+
+
+def process_taskrun_logs(
+    taskruns, core_v1_client, namespace, library, tekton_ci_output, check_net_http
+):
+    for tr in taskruns:
+        pod_name = tr["status"]["podName"]
+        taskrun_name = tr["metadata"]["name"]
+        logs = core_v1_client.read_namespaced_pod_log(
+            pod_name, namespace, container="step-unittest"
+        )
+        if "Installing" in logs:
+            print(
+                f"Retrieving container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}.."
+            )
+            if check_net_http:
+                version_match = re.search("(net-http) \([\w:]* ([\d.]+)\)", logs)
+                tekton_ci_output += f"{version_match[1]} {version_match[2]}\n"
+                check_net_http=False
+
+            match = re.search(f"Installing ({library} [^\s]+)", logs)
+            tekton_ci_output += f"{match[1]}\n"
+            break
+        else:
```

Comment:
I haven't caught this in the python scenario, but could this branch also be an early return instead?

Reply:
I think we cannot do an early return here because we need to get the next successful taskrun's logs if the latest successful taskrun's logs are unavailable.

Reply:
Sorry I meant, an early `continue`.

Reply:
Done!